core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -217,13 +220,35 @@ public LogicalPlan visitLimit(Limit node, AnalysisContext context) {
   public LogicalPlan visitFilter(Filter node, AnalysisContext context) {
     LogicalPlan child = node.getChild().get(0).accept(this, context);
     Expression condition = expressionAnalyzer.analyze(node.getCondition(), context);
+    verifySupportsCondition(condition);
 
     ExpressionReferenceOptimizer optimizer =
         new ExpressionReferenceOptimizer(expressionAnalyzer.getRepository(), child);
     Expression optimized = optimizer.optimize(condition, context);
     return new LogicalFilter(child, optimized);
   }
 
+  /**
+   * Ensure NESTED function is not used in WHERE clause. Fallback to legacy engine. Can remove when
+   * support is added for NESTED function in WHERE and GROUP BY clauses.
```

Comment:
Is there a github ticket where we will be adding nested in WHERE and GROUP BY clauses?

Reply:
Not yet no.